addons/eventbus/src/main/java/org/vaadin/spring/events/internal/MethodListenerWrapper.java

```diff
@@ -98,11 +98,14 @@ public boolean supports(Event<?> event) {
             if (listenerMethod.isAnnotationPresent(EventBusListenerMethod.class)) {
                 EventBusListenerMethod annotation = listenerMethod.getAnnotation(EventBusListenerMethod.class);
                 EventBusListenerMethodFilter filter = annotation.filter().newInstance();
+                String target = annotation.target();
                 EventScope scope = annotation.scope();
                 if (scope.equals(EventScope.UNDEFINED)) {
                     scope = event.getScope();
                 }
-                supports = supports && filter.filter(event.getPayload()) && event.getScope().equals(scope);
+                supports = supports && filter.filter(event.getPayload())
+                        && event.getScope().equals(scope)
+                        && event.getTarget().startsWith(target);
```

Comment:
So it's possible to publish an event with the target e.g. "org.vaadin.spring.event.first" and another one with "org.vaadin.spring.event.second". Both events could be handled by the same method with the taget "org.vaadin.event". An example can be seen in test class within the method "onIntegerPayloadEventWithTargetFail".

So there is the possibility to get something like a hierachy for the targets. It's not important yet, but can be usefull in the future.